noxfile.py

```diff
@@ -11,11 +13,38 @@
 from noxconfig import PROJECT_CONFIG
 
 
+class TestSet(Enum):
+    GPU_ONLY = "gpu-only"
+    DEFAULT = "default"
+
+
 @nox.session(name="integration-test-list", python=False)
 def run_integration_test_list(session: nox.Session):
     """
     Returns the test files under directory test (without walking subdirectories) as JSON list.
     """
+    test_set_values = [ts.value for ts in TestSet]
+    parser = ArgumentParser(
+        usage=f"nox -s {session.name} -- --test-set {{{','.join(test_set_values)}}}"
+    )
+    parser.add_argument(
+        "--test-set",
+        type=TestSet,
+        required=True,
+        help="Test set name",
+    )
+    args = parser.parse_args(session.posargs)
     test_path = PROJECT_CONFIG.root / "test"
-    tests = [{"path": str(t), "name": t.stem} for t in test_path.glob("test_*.py")]
+    if args.test_set == TestSet.GPU_ONLY:
+        tests = [
+            {"path": str(t), "name": t.stem}
+            for t in test_path.glob("test_*.py")
+            if "gpu" in t.name
```

Comment:
Not nice, but I propose to use later a similar solution as in ITDE after we migrated the tests to `pytest`